cirq/google/engine/engine.py

```diff
@@ -263,18 +255,13 @@ def program_as_schedule(self, program: TProgram) -> schedules.Schedule:
 
         raise TypeError('Unexpected program type.')
 
-    def run_sweep(
-            self,
-            *,  # Force keyword args.
-            program: TProgram,
-            program_id: Optional[str] = None,
-            job_config: Optional[JobConfig] = None,
-            params: study.Sweepable = None,
-            repetitions: int = 1,
-            priority: int = 500,
-            processor_ids: Sequence[str] = ('xmonsim',),
-            gate_set: serializable_gate_set.SerializableGateSet = gate_sets.XMON
-    ) -> engine_job.EngineJob:
+    def run_sweep(self,
+                  program: Union[circuits.Circuit, schedules.Schedule],
+                  params: study.Sweepable,
+                  repetitions: int = 1,
+                  program_id: Optional[str] = None,
+                  job_config: Optional[JobConfig] = None
```

Comment:
Is it okay to have the additional keyword arguments `program_id` and `job_config` even though they are not present in `Sampler.run_sweep`? I don't think it makes sense to set these in `__init__` because if they are both set to non-None values then they need to be changed for every run.

Reply:
I would prefer for us to stick to the interface exactly... @dabacon thoughts?

Reply:
I would also prefer to stick to the interface; it would be good to check whether mypy (and abc and...) still considers a valid "implementation" if there are extra params like this.

I've stumbled on this as well with some internal implementations of `Sampler` where we need extra per-call information that is implementation-specific. Could we add an extra argument in the `Sampler` interface to allow passing extra info which implementations can use as needed?

Reply:
Wouldn't that harm the interchangeability? E.g. one of the really nice things about EngineSampler is that I can substitute `cirq.Simulator` with absolutely no other changes. If EngineSampler took some additional arguments, that would not be the case.